src/cockpit/channels/stream.py

```diff
@@ -85,11 +86,17 @@ def create_transport(self, loop: asyncio.AbstractEventLoop, options: Dict[str, A
             if not isinstance(environ, list) or not all(isinstance(e, str) and '=' in e for e in environ):
                 raise ChannelError('protocol-error', message='invalid "environ" option for stream channel')
 
-            env.update({n: v for n, v in (envline.split('=', 1) for envline in environ)})
+            userenv = {n: v for n, v in (envline.split('=', 1) for envline in environ)}
```

Comment:
Unnecessary comprehension.  This looks better like:

```python
userenv = dict(e.split('=', 1) for e in environ)
```

Reply:
Done!